js/src/logger.ts

```diff
@@ -2878,6 +2886,13 @@ export class SpanImpl implements Span {
     return this._id;
   }
 
+  public setAttributes(args: Omit<StartSpanArgs, "event">): void {
+    this.internalData.span_attributes = {
```

Comment:
I realize this is confusing, but `internalData` is cleared immediately after it is set because we call `log` immediately after every time. Similarly, this method should probably trigger a `log` otherwise the span attributes may not get logged.

But better yet would be to make a minor refactor to something like this
```
public log(event: ExperimentLogPartialArgs): void {
  this.logInternal({ event });
}

private logInternal({ event, internalData }: { event: ExperimentLogPartialArgs: internalData?: Partial<ExperimentEvent> }) {
  // Basically the current implementation of `log`.
}
```

And then we can get rid of the `internalData` member variable and just call `logInternal` instead.

Reply:
That seems to work? Can you take a look at the code change and confirm I accomplished your suggestion?